GoogleForCreators/web-stories-wp
# Right Click Menu: Foreground Media Menu

## Context

Right Click Menu work. [Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7347%3A60835)

## Summary

Adding the foreground image menu. Right clicking on any media that is not a background image will open this menu.

## Relevant Technical Choices



## To-do

- [x] Finish actions
- [x] Fix old failing tests 😭 

## User-facing changes

|Forward and Back Actions|Setting Page Background|Scale and Crop|
|--|--|--|
|![forward-and-back](https://user-images.githubusercontent.com/22185279/126533973-ed5cefea-0358-4350-9b70-b83408d75d51.gif)|![page-background](https://user-images.githubusercontent.com/22185279/126533781-d6207de2-9246-4790-9861-da332790cad4.gif)|![scale-and-crop](https://user-images.githubusercontent.com/22185279/126533855-6ec324da-4dd3-477b-bb48-dc86a3c6afe3.gif)|

Reading the menu using voiceover:
<img src="https://user-images.githubusercontent.com/22185279/126535297-c23a856f-562e-43db-8220-0f0396d43cc5.gif" height="350px" />

## Testing Instructions

1. Turn on the context menu experiment
2. Create a new story
3. Add several pieces of media to the canvas (images, videos, gifs, etc)
4. Right click on an image.
5. Should be able to move media layers forward and back. Should not be able to move it below the background layer. Should not be able to move above the top layer.
6. Should be able to set an image as a page background. Doing this again will remove the previous background from the canvas.
7. Should be able to open the scale & crop UI

### QA






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6136
